gtf2featureAnnotation.R

```diff
@@ -224,23 +237,31 @@ if ( opt$feature_type == 'transcript' &&  all(c('transcript_id', 'transcript_ver
     anno$transcript_id <- versioned_transcripts
   } 
 }
-
+save.image()
 # If specified, filter down a provided cDNA FASTA file
 
 if (! is.null(opt$parse_cdnas)){
   
   # Derive annotation table from transcripts and use to augment GTF where necessary
   
   print("Parsing annotation info from cDNA FASTA headers")
-  tinfo <- plyr::rbind.fill(lapply(names(cdna), parse_ensembl_fasta_transcript_info))
+  fa_header_style <- NULL
+  if ('source' %in% colnames(anno)){
+    if (any(grepl('ensembl', anno$source)) || grepl('description:', names(cdna)[1])){
+      fa_header_style <- 'ensembl' 
+    }else{
+      fa_header_style <- tolower(as.character(anno$source[1]))
```

Comment:
will fa_header_style here always be equal to `wormbase`?

Reply:
It will be equal to whatever is in the 'source' column. I originally used the 'source' directly, but then realised that some wormbase files in Ensembl still followed Ensembl standards in the header. So, where we see 'description:' in the header we flag it as Ensembl style irrespective of source.